framework/include/variables/MooseVariableScalar.h

```diff
@@ -141,6 +141,11 @@ class MooseVariableScalar : public MooseVariableBase
 
   void insert(NumericVector<Number> & soln);
 
+  /**
+   * The highest solution state required for this variable (0 = current, 1 = old, etc...).
+   */
+  unsigned int needSolutionState() const;
+
```

Comment:
Could we rename this to `needHighestSolutionState`?

Reply:
`oldestSolutionState` / `oldestSolutionStateNeeded` ?